dpppt-backend-sdk/dpppt-backend-sdk-ws/src/main/java/org/dpppt/backend/sdk/ws/util/ValidationUtils.java

```diff
@@ -41,13 +41,9 @@ public boolean isValidBase64Key(String value) {
 	}
 
 	public boolean isDateInRange(OffsetDateTime timestamp) {
-		if (timestamp.isAfter(Instant.now().atOffset(ZoneOffset.UTC))) {
-			return false;
-		}
-		if (timestamp.isBefore(Instant.now().atOffset(ZoneOffset.UTC).minus(retentionPeriod))) {
-			return false;
-		}
-		return true;
+		OffsetDateTime now = Instant.now().atOffset(ZoneOffset.UTC);
+		OffsetDateTime retention = now.minus(retentionPeriod);
+		return timestamp.isAfter(retention) && timestamp.isBefore(now);
```

Comment:
@UBaggeler I'm not sure if this behaviour leads to problems, since retention and now are not included? 

`timestamp.isBefore(now) => timestamp < now` but `batchReleaseTime (aka timestamp) > now => timestamp <= now`

Reply:
It is true that we're missing the equality with `now`. However, as `now` is shifting anyway in the different parts of the verifications, I put this into `won't hurt`. According to java-documentation, `now` has millisecond resolution, so in the worst case, this comparison rejects a timestamp one millisecond too early.

Where it _does_ matter is if the comparison is done with computed values, like `atStartOfDay`. Then it is important to do a real `isAfterOrEqual`.

Reply:
Added a comment to point out this is at most `1ms` off. As the `now` is not cleanly defined anyway, I prefer readability over precision that is not available anyway.